File: leetcode/september-challenge-2021/shifting-letters.cpp
/* https://leetcode.com/problems/shifting-letters/ */

class Solution {
public:
    string shiftingLetters(string s, vector<int>& shifts) {
        int len = s.size();
        int sum = 0;

        for (int i = len - 1; i >= 0; i--) {
            sum += shifts[i];
            sum %= 26;
            s[i] = (char)(((int)s[i] + sum - 97) % 26 + 97);
        }

        return s;
    }
};

File: leetcode/september-challenge-2021/shifting-letters.js
/* https://leetcode.com/problems/shifting-letters/ */

/**
 * @param {string} s
 * @param {number[]} shifts
 * @return {string}
 */
 var shiftingLetters = function(s, shifts) {
    const len = s.length;
    let sum = 0;
    let str = s.split('');

    for (let i = len - 1; i >= 0; i--) {
        sum += shifts[i];
        sum %= 26;
        str[i] = String.fromCharCode((s[i].charCodeAt(0) + sum - 97) % 26 + 97);
    }

    return str.join('');
};
